tests/testsuite/support/sandbox.rs

```diff
@@ -0,0 +1,505 @@
+    }
+
+    /// Setup mock to return the latest version of yarn (chainable)
+    pub fn yarn_latest(mut self, version: &str) -> Self {
+        let mock = mock(method_name("GET"), "/yarn-latest")
+            .with_status(200)
+            .with_body(version)
+            .create();
+        self.root.mocks.push(mock);
+        self
+    }
+
+    /// Setup mocks to return info about the yarn archive file (chainable)
+    pub fn yarn_archive_mocks(mut self) -> Self {
+        // ISSUE(#145): this should actually use a real http server instead of these mocks
+
+        // generate a "file" that is 200 bytes long
+        let mut rng = thread_rng();
+        let archive_file_mock: String = iter::repeat(())
+            .map(|()| rng.sample(Alphanumeric))
+            .take(200)
+            .collect();
+
+        // mock the HEAD request, which gets the file size
+        let head_mock = mock(method_name("HEAD"), Matcher::Regex(r"^/yarn-v\d+.\d+.\d+".to_string()))
+            .with_header("Accept-Ranges", "bytes")
+            .with_body(&archive_file_mock)
+            .create();
+        self.root.mocks.push(head_mock);
+
+        // mock the "Range: bytes" request, which gets the ISIZE value (last 4 bytes)
+        // this will be interpreted as a packed integer value
+        // (doesn't really matter - used for progress bar)
+        let range_mock = mock(method_name("GET"), Matcher::Regex(r"^/yarn-v\d+.\d+.\d+".to_string()))
+            .match_header("Range", Matcher::Any)
+            .with_body("1234")
+            .create();
+        self.root.mocks.push(range_mock);
+
+        // mock the file download
+        let file_mock = mock(method_name("GET"), Matcher::Regex(r"^/yarn-v\d+.\d+.\d+".to_string()))
+            .match_header("Range", Matcher::Missing)
+            .with_body(&archive_file_mock)
+            .create();
+        self.root.mocks.push(file_mock);
+
+        self
+    }
+
+    /// Create the project
+    pub fn build(mut self) -> Sandbox {
+        // First, clean the directory if it already exists
+        self.rm_root();
+
+        // Create the empty directory
+        self.root.root().mkdir_p();
+
+        // make sure these directories exist
+        t!(fs::create_dir_all(node_cache_dir()));
+        t!(fs::create_dir_all(yarn_cache_dir()));
+        t!(fs::create_dir_all(notion_tmp_dir()));
+
+        // write node and yarn caches
+        for cache in self.caches.iter() {
+            cache.build();
+        }
+
+        // write files
+        for file_builder in self.files {
+            file_builder.build();
+        }
+
+        // join dirs for the path (notion bin path is already first)
+        self.root.path = env::join_paths(self.path_dirs.iter()).unwrap();
+
+        let SandboxBuilder { root, .. } = self;
+        root
+    }
+
+    fn rm_root(&self) {
+        self.root.root().rm_rf()
+    }
+}
+
+// files and dirs in the sandbox
+// TODO: some of these are different on windows - how to sandbox that?
```

Comment:
Don't leave any TODO comments (file issues if they're still open issues).

Reply:
just removed this line, I already took care of the todo and forgot to delete it